### DISCUSSION TITLE
My app.css file is empty laravel 10

### DISCUSSION THREAD

**Discussion by gurds91** [laravel/framework#49085] [Q&A]
Hi guys,

I am using laravel version 10.31 

my app.css file is empty, I tried running various commands in bash:

composer require laravel/ui

php artisan ui bootstrap

I installed node.js then ran commands npm install and npm run dev

Still nothing in my app.css file, in my app.scss file I do have this though:

// Fonts
@import url('https://fonts.bunny.net/css?family=Nunito');

// Variables
@import 'variables';

// Bootstrap
@import 'bootstrap/scss/bootstrap';

Any help what to do to get my app.css file compiled with code?

**Comment by antimech** [comment]
Mix is now replaced with Vite. When running `npm run dev` you can find your `scss` file here:
http://127.0.0.1:5173/resources/sass/app.scss

Run this if you want to build your frontend for production:
```Shell
npm run build
```
Then you could find your `css` file in `public/build/assets/`.

By the way, Laravel UI is now a legacy package, you should consider using [Laravel Breeze](https://github.com/laravel/breeze) for new projects. Or, for something more robust, consider [Laravel Jetstream](https://github.com/laravel/jetstream).

**Reply by gurds91** [reply]
hi , is this correct?

I looked at the files you posted and in my public/resources/js area I created a new file called vite.config.js and copied and pasted code into there from he vite.config.js file you sent.

I then ran command npm run build, a css file was created in public/build folder, in the terminal it has 3 problems - I attached screenshot.

Though the css file in my resources area , ,app.css is still empty? Is this a problem seeing as a css file was created in public/build?

![css file in build](https://github.com/laravel/framework/assets/107938466/ff1787c0-7251-4aea-af82-8364deb7b55d)

**Reply by antimech** [reply]
After you install Laravel UI with Boostrap preset:
* `resources/sass/app.scss` file will be added, which contains [this](https://github.com/laravel/ui/blob/4.x/src/Presets/bootstrap-stubs/app.scss);
* `vite.config.js` will be [reconfigured](https://github.com/laravel/ui/blob/4.x/src/Presets/bootstrap-stubs/vite.config.js) to compile `resources/sass/app.scss` instead of `resources/css/app.css`, here is [how the file looked before installation](https://github.com/laravel/laravel/blob/10.x/vite.config.js);
* And [other stuff](https://github.com/laravel/ui/blob/4.x/src/Presets/Bootstrap.php).

Just delete `resources/css/app.css` now it doesn't do anything, just confuses you. To make it clear what is actually happening after your actions consider creating local git repository for your code and commit often.

**Reply by gurds91** [reply]
Where is the vite.config.js file located? I created this file myself and copied and pasted code from here 

I ran PHP artisan ui bootstrap and I have the .scss file you are talking about.

**Reply by antimech** [reply]
In the root directory of your project, delete the other one you created.

**Reply by gurds91** [reply]
ok, I entered the command in bash , php artisan ui bootstrap --auth to do get the laravel ui with bootstrap preset, this is correct?  Then I entered npm install and npm run dev.

So now I can just delete my app.css file in resources/css?

**Reply by gurds91** [reply]
when I ran php artisan ui bootstrap command effectively it created a layout in the views section:

<!doctype html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    
    <meta name="csrf-token" content="{{ csrf_token() }}">

    <title>{{ config('app.name', 'Laravel') }}</title>

    
    <link rel="dns-prefetch" href="//fonts.bunny.net">
    <link href="https://fonts.bunny.net/css?family=Nunito" rel="stylesheet">

    
    @vite(['resources/sass/app.scss', 'resources/js/app.js'])
</head>
<body>
    <div id="app">
        <nav class="navbar navbar-expand-md navbar-light bg-white shadow-sm">
            <div class="container">
                <a class="navbar-brand" href="{{ url('/') }}">
                    {{ config('app.name', 'Laravel') }}
                </a>
                <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="{{ __('Toggle navigation') }}">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <div class="collapse navbar-collapse" id="navbarSupportedContent">
                    
                    <ul class="navbar-nav me-auto">

                    </ul>

                    
                    <ul class="navbar-nav ms-auto">
                        
                        @guest
                            @if (Route::has('login'))
                                <li class="nav-item">
                                    <a class="nav-link" href="{{ route('login') }}">{{ __('Login') }}</a>
                                </li>
                            @endif

                            @if (Route::has('register'))
                                <li class="nav-item">
                                    <a class="nav-link" href="{{ route('register') }}">{{ __('Register') }}</a>
                                </li>
                            @endif
                        @else
                            <li class="nav-item dropdown">
                                <a id="navbarDropdown" class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false" v-pre>
                                    {{ Auth::user()->name }}
                                </a>

                                <div class="dropdown-menu dropdown-menu-end" aria-labelledby="navbarDropdown">
                                    <a class="dropdown-item" href="{{ route('logout') }}"
                                       onclick="event.preventDefault();
                                                     document.getElementById('logout-form').submit();">
                                        {{ __('Logout') }}
                                    </a>

                                    <form id="logout-form" action="{{ route('logout') }}" method="POST" class="d-none">
                                        @csrf
                                    </form>
                                </div>
                            </li>
                        @endguest
                    </ul>
                </div>
            </div>
        </nav>

        <main class="py-4">
            @yield('content')
        </main>
    </div>
</body>
</html>

**Reply by antimech** [reply]
> ok, I entered the command in bash , php artisan ui bootstrap --auth to do get the laravel ui with bootstrap preset, this is correct? Then I entered npm install and npm run dev.

Yes. Why not?

> So now I can just delete my app.css file in resources/css?

Like I said.
